expression_json.go

```diff
@@ -31,6 +31,26 @@ func parseJsonField(data []interface{}) (Expression, error) {
 	return out, nil
 }
 
+func parseJsonFunc(data []interface{}) (Expression, error) {
+	var out FuncExpr
+	pos := 1
+	if funcName, ok := data[pos].(string); ok {
```

Comment:
the method seems to make a lot of assumptions.
1. data is of size at least 2.
2. data[1] may be a string, in which case it is always func name
3. data[1] may not be string, in which cases it is always a func parameter

is it possible that data does not contain func name, but the first parameter is of string type?

Reply:
Whilst your right that we should probably be performing length checks, we don't yet do this in any other method, so I didn't bother to include it.  I've filed #41 so we can go back and clean up the JSON parser to include the proper length/type checking.

Reply:
Ahh.  The JSON format explicitly defines the type of expression first.  For instance, a full function definition might look like this:
```
["equals",
  ["func",
    "mathRound",
    ["field", "latitude"]
  ],
  ["value", 37]
]
```

Reply:
so the assumption is that func name is always present? if so we might as well remove the ok check.

Reply:
It shouldn't really be assuming anything, since malformed data could be passed in, but that will be fixed in #41, for now its no worse than it is for any of the other parsers.